File: in_Java/0485 Max Consecutive Ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        if(nums == null || nums.length == 0) return 0;
        int res = 0;
        int i = 0;
        while(i < nums.length && nums[i] != 1) i++;
        int id = i;
        while(id < nums.length){
            while(i < nums.length && nums[i] == 1){
                i++;
            }
            if(i - id > res) res = i - id;
            id = i+1;
            i++;
        }
        return res;        
    }
}

File: in_Python/0485 Max Consecutive Ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = 0
        start = 0
        while start < len(nums):
            while start < len(nums) and nums[start] == 0:
                start = start + 1
            end = start
            while end < len(nums) and nums[end] == 1:
                end = end + 1
            res = max(res, end - start)
            start = end
        return res